src/dataflow-types/src/client/controller/compute.rs

```diff
@@ -123,7 +123,12 @@ impl<'a, T: Timestamp + Lattice> ComputeController<'a, T> {
     }
 }
 
-impl<'a, C: Client<T>, T: Timestamp + Lattice> ComputeControllerMut<'a, C, T> {
+impl<'a, C: Client<T>, T> ComputeControllerMut<'a, C, T>
+where
+    T: Timestamp + Lattice + TryInto<i64> + TryFrom<i64>,
+    <T as TryInto<i64>>::Error: std::fmt::Debug,
+    <T as TryFrom<i64>>::Error: std::fmt::Debug,
+{
```

Comment:
Mentioned elsewhere, but these constraints seem like noise. I think if you put them on an `impl` block for the (I believe) one method that requires them, then no one else should need them.